code/modules/admin/verbs/onlyone.dm

```diff
@@ -1,12 +1,86 @@
-		if(S.isDead() || !S.client) continue
-		if(is_special_character(S)) continue
+	var/list/pickfrom = list("Cancel")
+	for(var/datum_type in typesof(/datum/only_one))
+		var/datum/only_one/D = datum_type //For the undocumented but super cool initial() behavior
+		var/name = initial(D.name)
+		if(name)
+			pickfrom += name
+			pickfrom[name] = D
 
+	var/event_type_name = input(user, "Select an event.", "THERE CAN BE ONLY ONE", null) in pickfrom as text|null
+	if(!event_type_name)
+		return
+	var/event_type = pickfrom[event_type_name]
+	if(!event_type)
+		return
+
+	var/datum/only_one/event = new event_type()
+
+	if(!event.event_setup_start())
+		message_admins("<span class='notice'>[key_name_admin(usr)] tried to use THERE CAN BE ONLY ONE, but it failed in setup. (Type: [event_type_name])</span>", 1)
+		log_admin("[key_name(usr)] tried to use there can be only one, but it failed in setup. (Type: [event_type_name])")
+		return
+
+	message_admins("<span class='notice'>[key_name_admin(usr)] used THERE CAN BE ONLY ONE! (Type: [event_type_name])</span>", 1)
+	log_admin("[key_name(usr)] used there can be only one. (Type: [event_type_name])")
+
+	var/list/mobs_to_convert = list()
+	for(var/mob/M in event.eligible_mobs)
+		if(event.check_eligibility(M))
+			mobs_to_convert += M
+
+	for(var/mob/M in mobs_to_convert)
+		event.convert_mob(M)
+
+	event.event_setup_end()
+
+	qdel(event)
+
+
+/datum/only_one
+	var/name = "" //What to show in the list of choices. Does not appear in the list if left blank.
+	var/list/eligible_mobs //List of mobs to CHECK for eligibility. Not the final list of mobs to convert. Assigned in New() because BYOND a shit.
+
+//Only used for assigning eligible_mobs at the moment. It doesn't work in the type definition.
+/datum/only_one/New()
+	eligible_mobs = player_list
+
+//Called at the beginning of the event. Cancels the event if it returns false, so use this to start anything you want to start immediately and/or check if the event can't happen.
+/datum/only_one/proc/event_setup_start()
+	return 1
+
+//Used to retrieve the list of mobs to convert. Called on each mob in eligible_mobs. Any mob this returns true for is added to the final conversion list.
+/datum/only_one/proc/check_eligibility(var/mob/M)
+	if(M.stat == DEAD) //I don't use isDead() because that also returns 1 for fakedeath. Don't want to fuck over those in fakedeath when bussing.
+		return 0
+	if(ishuman(M) || issilicon(M))
+		return 1
+
+//Does the actual converting of mobs. Called on each mob in the final conversion list. (The base version of this proc doesn't do anything.)
+/datum/only_one/proc/convert_mob(var/mob/M)
+
+//Called at the end of the event setup.
+/datum/only_one/proc/event_setup_end()
+
+
+//The classic.
+/datum/only_one/highlander
+	name = "Highlander"
+
+/datum/only_one/highlander/check_eligibility(var/mob/M)
+	if(!..())
+		return 0
+	if(is_special_character(M))
```

Comment:
Yes, but that's how it was before. I think it's dumb, but I intentionally didn't functionally change anything.